images/gitlab-delete-stale-branches/delete_stale_branches.py

```diff
@@ -0,0 +1,106 @@
+#!/usr/bin/env python
+
+import json
+import os
+import re
+import urllib.parse
+from datetime import datetime, timedelta, timezone
+
+import requests
+import sentry_sdk
+
+sentry_sdk.init(
+    # This cron job only runs once a week,
+    # so just record all transactions.
+    traces_sample_rate=1.0,
+)
+
+
+GITLAB_API_URL = "https://gitlab.spack.io/api/v4/projects/2"
+AUTH_HEADER = {
+    "PRIVATE-TOKEN": os.environ.get("GITLAB_TOKEN", None)
+}
+
+
+def paginate(query_url):
+    """Helper method to get all pages of paginated query results"""
+    results = []
+
+    while query_url:
+        resp = requests.get(query_url, headers=AUTH_HEADER)
+
+        if resp.status_code == 401:
+            print(" !!! Unauthorized to make request, check GITLAB_TOKEN !!!")
+            return []
```

Comment:
I would also suggest raising an exception here instead of silently failing with an empty list -
```suggestion
        resp = requests.get(query_url, headers=AUTH_HEADER)

        if resp.status_code == 401:
            print(" !!! Unauthorized to make request, check GITLAB_TOKEN !!!")
            resp.raise_for_status()
```

That would give us a sentry alert if the gitlab token doesn't work (if it expires, for example). Basically I think we should want these internal services to fail as loudly as possible so that we can find out as soon as possible. I could see a scenario where it starts to fail and no one notices for a while because you'd need to be checking the pod logs (or notice the branches on gitlab growing).